openwisp_monitoring/monitoring/base/models.py

```diff
@@ -451,7 +428,7 @@ def _get_top_fields(self, number, time=DEFAULT_TIME, timezone=settings.TIME_ZONE
         q = self.get_query(
             query=q, summary=True, fields=['SUM(*)'], time=time, timezone=timezone
         )
-        res = list(query(q, epoch='s').get_points())
+        res = list(TimeseriesDB.query(q, epoch='s').get_points())
```

Comment:
`epoch='s').get_points()` it's influxdb specific.

get_points is a method of the influxdb client library, and is being called from models, hence models is still aware of inflxudb specific code, so the abstraction is not there yet. Abstraction means that `models.py` must not know anything about influxdb It must only call one method of the timeseries DB class, it's this class that will know what to do with the call.

Eg:

```python
timeseries_db.get(q)
```

Applies to all other occurrences, please fix this.